tools/normalize.c

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright 2020 Codethink Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <inttypes.h>
+#include <string.h>
+
+#include "../src/msg.h"
+
+#include "msg.h"
+#include "sig.h"
+#include "fifo.h"
+#include "util.h"
+
+// DEFAULT_AVERAGE_WIDTH, the number of milliseconds to average over
+long DEFAULT_AVERAGE_WIDTH = 1000;
+
+struct channel_data {
+	bl_msg_sample_data_t msg; // message to buffer output
+	struct fifo *samples; // all the samples in the rolling sum
+	uint64_t rolling_sum; // sum across the samples that will be averaged
+	uint32_t baseline; // "zero" level for an "average" sample
+};
+
+void add_normalized_sample(struct channel_data *channel, uint32_t sample)
```

Comment:
Minor nit: Could just pass the `bl_msg_sample_data_t *msg` in here, instead of the whole channel structure; makes it a bit clearer that it's adding sample to the message.